src/tty.rs

```diff
@@ -186,9 +186,20 @@ pub fn new<T: ToWinsize>(config: &Config, size: T) -> Pty {
 
     let (master, slave) = openpty(win.ws_row as _, win.ws_col as _);
 
-    let shell = config.shell().unwrap_or(pw.shell);
+    let (program, args) = match config.shell() {
```

Comment:
With changes to `Shell` definition, this block can be simplified:

```rust
let shell = config.shell().unwrap_or_else(|| Shell::new(pw.shell));
```

I guess `Shell` would need a `new` method as well.

Once the `-e` argument lands, the shell can be chosen from one of /etc/passwd, configuration, or the CLI like so:

```rust
let shell = options.shell()
    .or_else(|| config.shell())
    .unwrap_or_else(|| Shell::new(pw.shell));
```